eval/eval.go

```diff
@@ -925,6 +925,7 @@ func (e *evalContext) evaluateBuiltinFromJSON(x *expr, repr *fromJSONExpr) *valu
 			v.unknown = true
 			return v
 		}
+		ev.Secret = v.secret
```

Comment:
will this be sufficient? I'm worried that this needs to be deeply plumbed through `FromJSON`...

Reply:
Good point, fixed now.